### ISSUE TITLE
Consider using a different library for webhooks

### ISSUE THREAD

**Issue by Bibo-Joshi** [python-telegram-bot/python-telegram-bot#2966]
Currently we're using `tornado`. While it works, there are more modern libraries that may be more suitable for working in combination with `asyncio`. A possible condidate may e.g. be `starlette`.

We should evaluate the options and decided if this is something we want to do.

**Comment by tsnoam** [comment]
@Bibo-Joshi <code>aiohttp</code> is another viable option

**Comment by Dotrox** [comment]
I think the best choice would be to make the webhooks library pluggable, so everyone could use a library that suit their needs.
For example, I use Falcon and have spent a lot of time to making an external webhooks engine with it. But now I plan upgrade to v20 and will have to do it again.

**Comment by Bibo-Joshi** [comment]
@Dotrox Hi. in v20 it&#x27;s rather easy to use a custom webhook setup and this is showcased in <a href="https://docs.python-telegram-bot.org/examples.html#examples-customwebhookbot">customwebhookbot.py</a>. We still want to provide a built-in solution for webhooks.

**Comment by Bibo-Joshi** [comment]
Some time has passed since we published v20 and so far we have seen no issues with using tornado as framework for the built-in webserver. As we don't have any strong arguments to switch to any of the alternatives, I'll close this issue as won't do for now.

This does ofc not rule out that we may switch to a different framework in the future if either we find reasons for not using tornado or strong reasons for a specific other framework. While many of the asyncio-friendly frameworks advertise themself as very fast, I haven't looked for any benchmarks so far and maybe a ptb-specific benchmark would make even more sense.
For completeness, let me mention that `tornado` is dependency free, which is not the case for any of the other frameworks that I have looked at so far, which is a plus for `tornado` IMO.
